x-pack/docs/en/security/authorization/mapping-roles.asciidoc

```diff
@@ -2,17 +2,34 @@
 [[mapping-roles]]
 === Mapping users and groups to roles
 
-If you authenticate users with the `native` or `file` realms, you can manage
-role assignment by using the <<managing-native-users,user management APIs>> or
-the <<users-command,users>> command-line tool respectively.
+Role mapping is supported by all realms except `native` and `file`.
 
-For other types of realms, you must create _role-mappings_ that define which
-roles should be assigned to each user based on their username, groups, or
-other metadata.
+TIP:  Those two realms assign roles directly to users, they do not support role mapping.
+Native realms use <<managing-native-users,user management APIs>>.
+File realms use <<roles-management-file,File-based role management>>.
+
+If using Role Mapping, there are two supported sources of role mapping rules,
+<<mapping-roles-api, Role Mapping API>> and <<mapping-roles-file, Role Mapping File>>.
+
+IMPORTANT: Role Mapping API is preferred over Role Mapping File.
+
+The PKI, LDAP, AD, Kerberos, OpenID Connect, JWT, and SAML realms support
+<<mapping-roles-api, Role Mapping API>>. Only PKI, LDAP, and AD realms
+support <<mapping-roles-file, Role Mapping File>>.
+
+Besides Role Mapping, the PKI, LDAP, AD, Kerberos, OpenID Connect, JWT, and
+SAML realms realms support <<authorization_realms,Delegated Authorization>> too.
+Role Mapping and Delegated Authorization cannot be used simultaneously in a realm.
+Pick the best option available for each realm based on your requirements.
+
+If you choose role mapping, you will have to create roles and _role-mappings_.
+Role mapping rules can based on realm name, realm type, username, groups,
+other user metadata, or combinations of those values.
 
 NOTE: When <<anonymous-access,anonymous access>> is enabled, the roles
 of the anonymous user are assigned to all the other users as well.
 
+There are two sources of role-mapping rules.
```

Comment:
I'd replace this paragraph with the suggestion below. Can't suggest changes to non-modified text. :-)

```suggestion
If there are role-mapping rules created through the API as well as a role mapping file, 
the rules are combined. 
It's possible for a single user to have some roles that were mapped through the API, 
and others assigned based on the role mapping file.
``````